org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/model/ModelViewer.java

```diff
@@ -153,7 +159,43 @@ public void dispose() {
 		content.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 		GridUtils.removeMargins(content);
 		
-		this.viewer = new TableViewer(content, SWT.SINGLE | SWT.BORDER | SWT.FULL_SELECTION);
+		this.viewer = createTableViewer(content);		
+		
+		this.typeEditor = new TypeEditor<>(this, content,  SWT.NONE);
+		typeEditor.setLayoutData(new GridData(GridData.FILL_BOTH));
+		GridUtils.setVisible(typeEditor, false);
+		
+		this.validationComposite = new Composite(content, SWT.NONE);
+		validationComposite.setLayout(new GridLayout(2, false));
+		validationComposite.setLayoutData(new GridData(SWT.FILL, SWT.BOTTOM, true, false));
+		
+		final Label error = new Label(validationComposite, SWT.NONE);
+		error.setImage(Activator.getImageDescriptor("icons/error.png").createImage());
+		error.setLayoutData(new GridData(SWT.LEFT, SWT.BOTTOM, false, false));
+		
+		this.validationMessage = new Label(validationComposite, SWT.WRAP);
+		validationMessage.setForeground(Display.getDefault().getSystemColor(SWT.COLOR_RED));
+		validationMessage.setLayoutData(new GridData(SWT.FILL, SWT.BOTTOM, true, false));
+		GridUtils.setVisible(validationComposite, false);
+
+		if (PageUtil.getPage()!=null) {
+			ISelection selection = PageUtil.getPage().getSelection();
+			processWorkbenchSelection(selection); // If model view is selected later but something it can process is the page selection...
+			
+			if (model==null) { // Go and look for the model on a view
+				final IViewPart part  = PageUtil.getPage().findView(ScanView.ID);
+				final Object    model = part!=null ? part.getAdapter(IScanPathModel.class) : null;
+				processObject(model);
+			}
+		}
+		
+		createActions();
+		
+		return (U)content;
+	}
+
+	private TableViewer createTableViewer(Composite content2) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "content2". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)